examples/user_input.rs

```diff
@@ -47,10 +51,67 @@ impl Default for App {
             input: String::new(),
             input_mode: InputMode::Normal,
             messages: Vec::new(),
+            cursor_in_input: 0,
         }
     }
 }
 
+impl App {
+    fn move_cursor_left(&mut self) {
+        if !self.is_cursor_leftmost() {
+            let cursor_moved_left = self.cursor_in_input - 1;
+            self.cursor_in_input = cursor_moved_left;
+        };
+    }
+
+    fn move_cursor_right(&mut self) {
+        if !self.is_cursor_rightmost() {
+            let cursor_moved_right = self.cursor_in_input + 1;
+            self.cursor_in_input = cursor_moved_right;
+        };
+    }
```

Comment:
Look into using `saturating_sub` and `saturating_add` for these functions. This along with other changes would eliminate the need for `is_cursor_leftmost` and `is_cursor_rightmost`.

Reply:
(and clamp)

Reply:
In the moment of writing saturating_add seems to be nightly only.

https://doc.rust-lang.org/std/intrinsics/fn.saturating_add.html

Would that be okay for an example.

Reply:
You should be able to use the `saturating_add` implemented on [usize](https://doc.rust-lang.org/std/primitive.usize.html#method.saturating_add).

Also as @joshka pointed out, you'll also need to `clamp` the value.

Reply:
Do you mean that I should clamp the cursor position so that if it is to the most left, at 0, then it should go to the rightmost, at the end of the input field ?

Reply:
No, I mean clamp it so that it doesn't set the cursor position past the width of the input field.

Reply:
Saturating_add  and saturating_sub are used instead of + 1/ -1.
Clamped is also preformed via num::clamp.